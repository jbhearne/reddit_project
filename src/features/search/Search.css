.search .field {
    width: 80%;
    height: 2rem;
    background-color: #f2e2ee ;
    color: black;
    border: solid #8f1e66  3px;
    font-size: 1.5rem;
    border-radius: 50px;
    margin-right: 4px;

}

.search button {
    position: relative;
    width: 2.5rem;
    height: 2.5rem;
    font-size: 1.5rem;
    background-color: #32833d;
    border: solid #8f1e66  3px;
    border-radius: 50%;
    animation-name: buttonA;
    animation-duration: 2s;
}

@keyframes buttonA {
    from {
        left: 100%;
    }

    to {
        left: 0%;
    }
}


@media only screen and (max-width: 600px) {
    .search .field {
        width: 75%;
        height: 1.2rem;
        background-color: #f2e2ee ;
        color: black;
        border: solid #8f1e66  3px;
        font-size: .75rem;
        border-radius: 50px;
        margin-right: 4px;
    
    }
    .search button {
        position: relative;
        top: .4rem;
        width: 1.8rem;
        height: 1.8rem;
        font-size: 1.5rem;
        background-color: #32833d;
        border: solid #8f1e66  3px;
        border-radius: 50%;
    }
}