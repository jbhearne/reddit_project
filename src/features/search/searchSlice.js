import { createSlice } from '@reduxjs/toolkit';


export const searchSlice = createSlice({
    name: 'search',
    initialState: '',
    reducers: {
        search: (state, action) => {
            state = action.payload;
        }
    }
})

export const selectSearch = (state) => state.search;
export const search = searchSlice.actions.search;

export default searchSlice.reducer;
