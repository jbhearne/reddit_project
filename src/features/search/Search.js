import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { selectSearch, setSearch } from './searchSlice';
import { selectPostsPrefix, setPostsUrl } from '../posts/postsSlice';

export function Search() {

    const search = useSelector(selectSearch);
    const postsPrefix = useSelector(selectPostsPrefix);
    const dispatch = useDispatch();

    const handleChange = (e) => {
        dispatch(setSearch(e.target.value));
    }

    const handleSearchClick = (e) => {
        e.preventDefault()
        const url = postsPrefix + '/search.json?q=' + search;  //need to use .json extension for api to work
        dispatch(setPostsUrl(url));
        dispatch(setSearch(''))
    }


    return (
        <form className='search' onSubmit={handleSearchClick}>  {/*onSubmit goes on <form> not <button> this caused some frustration*/}
            <input type='text' placeholder='Search' name='search' onChange={handleChange}  value={search}></input>
            <button><img className='search_img'></img></button>
        </form>
    )
}