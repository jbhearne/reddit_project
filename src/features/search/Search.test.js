import { useSelector, useDispatch } from "react-redux";
import { selectSearch } from './searchSlice';
import { Search } from './Search';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

jest.mock('react-redux');
jest.mock('./searchSlice');

describe('Search component', () => {
    let mockComponentSearch = 'searchTest';

    it('should render', () => {
        useSelector.mockImplementation(selection => {
            if (selection === selectSearch) {
                return mockComponentSearch;
            } else {
                return 'nope';
            };
        });
        useDispatch.mockReturnValue((dis) => dis);
        const component = renderer.create(
            <Router>
                <Search />
            </Router>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});