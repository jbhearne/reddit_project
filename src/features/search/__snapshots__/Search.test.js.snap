// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should render 1`] = `
<form
  className="search"
  onSubmit={[Function]}
>
    
  <input
    className="field"
    name="search"
    onChange={[Function]}
    placeholder="  Search"
    type="text"
    value="searchTest"
  />
  <button>
    <img
      className="search_img"
    />
  </button>
</form>
`;

exports[`Search component should render 2`] = `
<form
  className="search"
  onSubmit={[Function]}
>
    
  <input
    className="field"
    name="search"
    onChange={[Function]}
    placeholder="  Search"
    type="text"
    value="searchTest"
  />
  <button>
    <img
      className="search_img"
    />
  </button>
</form>
`;
